### PR TITLE
Using Clang on CI

### PR DESCRIPTION
**Pull request by payetvin**
This PR branched from the #960

### CODE REVIEW

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -105,11 +106,19 @@ jobs:
...
+      # if: ${{ matrix.ortools-xpress == 'ON' }}
       run: |
         git clone -b master https://${{ secrets.XPRS_USERNAME }}:${{ secrets.XPRS_TOKEN }}@github.com/rte-france/xpress-mp ${{ env.XPRESSDIR }}
         ls -la ${{ env.XPRESSDIR }}
 
+    - name: Configure compiler
+      run: |
+        if [ "${{ matrix.compiler }}" == "clang" ]; then
+          export CC=/usr/bin/clang
+          export CXX=/usr/bin/clang++
```
Comment (flomnes) [inline .github/workflows/ubuntu.yml]: Those env variables won't be available to the next steps.

See [here](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable).

File: .github/workflows/ubuntu.yml [hunk 2]
```diff
@@ -55,13 +55,14 @@ jobs:
     - name: ccache
       uses: hendrikmuhs/ccache-action@v1.2
       with:
-          key: xpress-${{ matrix.ortools-xpress }}-${{ matrix.os }}
+          key: xpress-${{ matrix.compiler }}-${{ matrix.os }}
```
Comment (flomnes) [inline .github/workflows/ubuntu.yml]: OK

File: .github/workflows/ubuntu.yml [hunk 3]
```diff
@@ -41,7 +41,7 @@ jobs:
       matrix:
         os: [ubuntu-20.04]
         test-platform: [ubuntu-20.04]
-        ortools-xpress: [ON, OFF]
+        compiler: [gcc, clang]
```
Comment (flomnes) [inline .github/workflows/ubuntu.yml]: Please keep `ortools-xpress`

**Review by flomnes — CHANGES_REQUESTED** [review]

File: .github/workflows/ubuntu.yml [hunk 4]
```diff
@@ -55,13 +55,14 @@ jobs:
...
       with:
-          key: xpress-${{ matrix.ortools-xpress }}-${{ matrix.os }}
+          key: xpress-${{ matrix.compiler }}-${{ matrix.os }}
 
     - name: Install libraries
       run: |
            sudo apt-get update
            sudo apt-get install uuid-dev libwxgtk3.0-gtk3-dev
            sudo apt-get install libboost-test-dev
+           sudo apt-get install clang
```
Comment (flomnes) [inline .github/workflows/ubuntu.yml]: Only install if required

File: .github/workflows/ubuntu.yml [hunk 5]
```diff
@@ -73,13 +74,13 @@ jobs:
...
             echo "ORTOOLS_URL=https://github.com/rte-france/or-tools/releases/download/v9.2-rte1.3/ortools_cxx_ubuntu-20.04_static_sirius_xprs.zip" >> $GITHUB_ENV
             echo "LABEL=-ortools-xpress" >> $GITHUB_ENV
-          else
-            echo "ORTOOLS_URL=https://github.com/rte-france/or-tools/releases/download/v9.2-rte1.3/ortools_cxx_ubuntu-20.04_static_sirius.zip" >> $GITHUB_ENV
-            echo "LABEL=" >> $GITHUB_ENV
-          fi
+          # else
+          #   echo "ORTOOLS_URL=https://github.com/rte-france/or-tools/releases/download/v9.2-rte1.3/ortools_cxx_ubuntu-20.04_static_sirius.zip" >> $GITHUB_ENV
+          #   echo "LABEL=" >> $GITHUB_ENV
+          # fi
```
Comment (flomnes) [inline .github/workflows/ubuntu.yml]: Keep this

**Review by flomnes — CHANGES_REQUESTED** [review]

File: .github/workflows/ubuntu.yml (hunk 1 above)
Comment (payetvin) [reply .github/workflows/ubuntu.yml:118]: It only needs to be there for configuration, it's stored in the Cmake cache after that

**Comment by JasonMarechal25** [comment]
There is conflict managing Ortools Xpress [on|off] on one hand and compiler [gcc|clang] on the other hand.

File: .gitmodules [hunk 6]
```diff
@@ -4,7 +4,7 @@
 [submodule "src/antares-deps"]
 	path = src/antares-deps
 	url = https://github.com/AntaresSimulatorTeam/antares-deps.git
-	branch = v2.0.2
+	branch = fix/clang-boost
```
Comment (flomnes) [inline .gitmodules]: It shouldn't be necessary to change antares-deps. Just pass `-DBUILD_BOOST` to cmake's configure command ?

File: .github/workflows/ubuntu.yml [hunk 7]
```diff
@@ -86,19 +90,32 @@ jobs:
...
+           fi
+
            cmake -B _build -S src \
            -DCMAKE_C_COMPILER_LAUNCHER=ccache \
            -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
            -DDEPS_INSTALL_DIR=${{github.workspace}}/rte-antares-deps-Release \
            -DCMAKE_BUILD_TYPE=release \
            -DBUILD_TESTING=ON \
            -DBUILD_not_system=OFF \
+           -DBUILD_BOOST=ON \
```
Comment (flomnes) [inline .github/workflows/ubuntu.yml:105]: This should not be required

**Pull request closed without merging**